lib/reducers/priceBook.js

```diff
@@ -0,0 +1,52 @@
+import XMLFilterWriter from '#xml/XMLFilterWriter.js';
+import { getPostfixFile } from '#tools/files.js';
+import { formatToFixed } from '#tools/logger.js';
+
+/**
+ * @param {Array<string>} files
+ * @param {Array<string>} allReducedProducts
+ * @param {Array<string>} nonMasters
+ * @param {number | false} [defaultAmount]
+ */
+export default async (files, allReducedProducts, nonMasters, defaultAmount) => {
+    const savedProducts = [];
+
+    for (let index = 0; index < files.length; index++) {
+        const priceBookFile = files[index];
+
+        const priceBookFilterWritter = new XMLFilterWriter(priceBookFile, getPostfixFile(priceBookFile))
+            .setName('Pricebook reducer-writer');
+
+        /**
+         * If it's the last file
+         */
+        if (index === files.length - 1) {
```

Comment:
you should write prices not only into last file.
Each different price book file belong to different currency, or different price book type (list or sale)
Price should be present in each list pricebook.
You can detect if pricebook is list by checking pricebooks>pricebook>header>parent tag. If it is not present or empty then it is list price.